### ISSUE TITLE
prisma deploy is stuck when deploying two fields with same name but different casing

### ISSUE THREAD

**Issue by HaiderMalik12** [prisma/prisma#2793]
**Describe the bug**

**To Reproduce**
1. Clone the repo [prisma-boilerplate](https://github.com/HaiderMalik12/hackernews-clone-graphql/tree/challenge-fetch-links)
2. Run the prisma deploy
3. Prisma will stuck on the `Apply changes`

**Expected behavior**
It should execute the prisma deploy successfully and download the schema from the Prisma service

**Screenshots**
![screen shot 2018-07-20 at 10 33 44 am](https://user-images.githubusercontent.com/16774378/42985257-16f104be-8c0a-11e8-8799-46b387e41d09.png)

**Versions (please complete the following information):**
 - OS: `OS X High Sierra`
 - `prisma` CLI:  `prisma/1.12.0 (darwin-x64) node-v8.9.4`
 - Prisma Server: `1.12.0`
 - `graphql-cli`, `prisma-binding`, `graphql-yoga`,  `node-basic-boilerplate`

**Comment by divyenduz** [comment]
When I try to reproduce this, I get a different error:-

```
Applying changes !
 ▸    java.sql.SQLSyntaxErrorException: (conn=14467) Duplicate column name 'createdAt'
 ▸    at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:163)
 ▸    at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:106)
 ▸    at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:235)
 ▸    at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:224)
 ▸    at org.mariadb.jdbc.MariaDbPreparedStatementClient.execute(MariaDbPreparedStatementClient.java:159)
 ▸    at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
 ▸    at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
 ▸    at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:39)
 ▸    at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)
 ▸    at slick.jdbc.Invoker.first(Invoker.scala:31)
 ▸    at slick.jdbc.Invoker.first$(Invoker.scala:30)
 ▸    at slick.jdbc.StatementInvoker.first(StatementInvoker.scala:16)
 ▸    at slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:52)
 ▸    at slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:51)
 ▸    at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:240)
 ▸    at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:240)
 ▸    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 ▸    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 ▸    at java.lang.Thread.run(Thread.java:748)
 ▸    Caused by: java.sql.SQLException: Duplicate column name 'createdAt'
 ▸    Query is: ALTER TABLE `public-test-introspection-random~database@dev`.`Link` ADD COLUMN `createdAt`
 ▸    datetime(3)  NOT NULL , ALGORITHM = INPLACE
 ▸    at org.mariadb.jdbc.internal.util.LogQueryTool.exceptionWithQuery(LogQueryTool.java:146)
 ▸    at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:217)
 ▸    at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:218)
 ▸    ... 15 more
```

Did you use introspection to generate the `datamodel.graphql`? Would it be possible for you to share your `postgres` schema? 

We are working on some [improvements](https://github.com/prismagraphql/prisma/pull/2741) regarding introspection.

**Comment by HaiderMalik12** [comment]
Thanks @marktani

**Comment by marktani** [comment]
Added to https://github.com/prismagraphql/prisma/issues/2793.

**Comment by selipso** [comment]
I'm also having problems with case-sensitive fields, but my error message is different:
```
prisma_1    | org.postgresql.util.PSQLException: ERROR: column User.usertype does not exist
prisma_1    |   Hint: Perhaps you meant to reference the column "User.userType".
prisma_1    |   Position: 69
prisma_1    | 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2433)
prisma_1    | 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2178)
prisma_1    | 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:306)
prisma_1    | 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
prisma_1    | 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
prisma_1    | 	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:155)
prisma_1    | 	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:144)
prisma_1    | 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
prisma_1    | 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
prisma_1    | 	at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:38)
prisma_1    | 	at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:21)
prisma_1    | 	at slick.jdbc.Invoker.foreach(Invoker.scala:47)
prisma_1    | 	at slick.jdbc.Invoker.foreach$(Invoker.scala:46)
prisma_1    | 	at slick.jdbc.StatementInvoker.foreach(StatementInvoker.scala:15)
prisma_1    | 	at slick.jdbc.StreamingInvokerAction.run(StreamingInvokerAction.scala:22)
prisma_1    | 	at slick.jdbc.StreamingInvokerAction.run$(StreamingInvokerAction.scala:20)
prisma_1    | 	at slick.jdbc.SQLActionBuilder$$anon$1.run(StaticQuery.scala:95)
prisma_1    | 	at slick.jdbc.SQLActionBuilder$$anon$1.run(StaticQuery.scala:95)
prisma_1    | 	at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
prisma_1    | 	at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
prisma_1    | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
prisma_1    | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
prisma_1    | 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by selipso** [comment]
Actually, looks like my issue had to do with a non-nullable enum field type. Posted a new issue related to this here: https://github.com/prismagraphql/prisma/issues/2922

**Comment by mavilein** [comment]
Thanks for reporting the issue and your thorough description! 🙏 This has been fixed in `1.18.1`.
